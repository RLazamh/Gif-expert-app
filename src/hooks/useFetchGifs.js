import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';

const useFetchGifs = (category) => {
    const [state, setState] = useState({
        data:[],
        loading: true
    });


    useEffect( () =>{

        getGifs (category)
        .then( (imgs) => {
            setState({
                data: imgs,
                loading: false
            })
        })
    }, [category])
    
    // useEffect(()=>{  
    //     getGifs(category)
    //     .then((imgs)=>setState({
    //         data: imgs,
    //         loading: false}));
    // },[category])
    // setTimeout(()=>{
    //     setState({
    //         data:[1,2,3,4],
    //         loading: false
    //     })
    // },3000)
    return state;

    
}

export default useFetchGifs
