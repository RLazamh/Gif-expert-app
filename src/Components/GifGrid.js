// import React, { useEffect, useState } from 'react'
import React from 'react'
import useFetchGifs from '../hooks/useFetchGifs'
import PropTypes from 'prop-types'
import GifGirdItem from './GifGirdItem';
// import { getGifs } from '../helpers/getGifs';

function GifGrid({category}) {
    const {loading, data: images } = useFetchGifs(category);
    // console.log(loading);
    // console.log(images);
    // el codigo que se ejecutara una sola vez se manda las dependencias y la funcion 
    // la funcion que tiene adentro se disparar una sola vez :)
    // const [image, setImage] = useState([]);
    return (
        <>
            <h2 className="animate__animated animate__fadeIn">{category}</h2>
            {loading && <p className="animate__animated animate__flash">Cargando.......</p>}
            <div className="card-grid ">
            
                {
                    images.map((img)=>(
                        <GifGirdItem 
                        key ={img.id} 
                        {...img}/>
                    ))
                }
              
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,

}

export default GifGrid;

