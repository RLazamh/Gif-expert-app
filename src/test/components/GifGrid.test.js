import { shallow } from 'enzyme'
import GifGrid from '../../Components/GifGrid'
import useFetchGifs from '../../hooks/useFetchGifs'
// para ayuda 
import '@testing-library/jest-dom';

jest.mock('../../hooks/useFetchGifs') // fingir useFetch 

describe('Pruebas en el componente <GifGrid />', () => {
    const category = 'Hola' 
     
    test('should match with snapshot ', () => {

       useFetchGifs.mockReturnValue({
            data:[],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();
    })

    test('should No show the <p /> ', () => {

        const gifs = [{
            id:'ABC',
            url:'https://loca.jpg',
            title: 'Cualquier cosa',

        },{
            id:'123',
            url:'https://loca.jpg',
            title: 'Cualquier cosa',
        }]

        useFetchGifs.mockReturnValue({
             data:gifs,
             loading: false
         });
         const wrapper = shallow(<GifGrid category={category}/>)
         expect(wrapper).toMatchSnapshot();

         expect(wrapper.find('p').exists()).toBe(false)
         expect(wrapper.find('GifGirdItem').length).toBe(gifs.length)
     })


    // test('should show items when carge iamge useFer ', () => {

    // })
    
    
    
})
