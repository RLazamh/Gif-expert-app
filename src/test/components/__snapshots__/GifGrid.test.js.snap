// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> should No show the <p />  1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Hola
  </h2>
  <div
    className="card-grid "
  >
    <GifGirdItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://loca.jpg"
    />
    <GifGirdItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://loca.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> should match with snapshot  1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Hola
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    Cargando.......
  </p>
  <div
    className="card-grid "
  />
</Fragment>
`;
